flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/common/matrix/DenseMatrixUtil.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.common.matrix;
+
+/**
+ * DenseMatrixUtil provides some operations over DenseMatrix.
+ */
+public class DenseMatrixUtil {
+
+	/**
+	 * Provides access to DenseMatrix's internal data buffer. Be aware that
+	 * the data is stored in column major.
+	 *
+	 * @param mat The matrix.
+	 * @return The matrix's internal data buffer.
+	 */
+	public static double[] getDataBuffer(DenseMatrix mat) {
+		return mat.data;
```

Comment:
wonder whether we should allow this. it does not return a copy of the internal array. If I create another DenseMatrix using `createFromBuffer` it seems like it is easy for users to accidentally touch the other one. 

I think at lease one of these should be a deep copy.

Reply:
We have removed this method.